pkg/serviceprovider/lookup.go

```diff
@@ -40,37 +40,63 @@ type GenericLookup struct {
 	MetadataProvider MetadataProvider
 	// MetadataCache is an abstraction used for storing/fetching the metadata of tokens
 	MetadataCache *MetadataCache
+	// RepoHostParser is a function that extracts the host from the repoUrl
+	RepoHostParser RepoHostParser
+}
+
+type RepoHostParser interface {
+	Host(url string) (string, error)
+}
+
+type RepoHostParserFunc func(repoUrl string) (string, error)
+
+func (f RepoHostParserFunc) Host(url string) (string, error) {
+	return f(url)
+}
+
+func RepoHostFromUrl(repoUrl string) (string, error) {
+	parsed, err := url.Parse(repoUrl)
```

Comment:
this won't work for quay without `https://`. We need to pass this test.

```
func TestHost(t *testing.T) {
	host, err := RepoHostFromUrl("https://quay.io/repository/redhat_emp1/spi-test")
	assert.Equal(t, "quay.io", host)
	assert.NoError(t, err)

	host, err = RepoHostFromUrl("quay.io/repository/redhat_emp1/spi-test")
	assert.Equal(t, "quay.io", host)
	assert.NoError(t, err)
}
```

Reply:
quay doesn't use this function directly though - https://github.com/redhat-appstudio/service-provider-integration-operator/pull/128/files#diff-277db106a268114e85ef6f6c87ee11d585b4802808027bc5407ba86696927838R71

Reply:
There is one other place where the repoUrl is parsed in quay impl: https://github.com/redhat-appstudio/service-provider-integration-operator/pull/128/files#diff-fc99dc356c1474c6a3dcdb4a27111aa03f2f513a9d452d9ba2d477e5b7c26e78R354 and that, too, should handle both the repo urls with and without scheme.

Reply:
I see. I've missed that because before it was called directly. Which brings back my first comment to this OOP thing, that it brings more complexity than it makes it easier. I don't like this approach in Go.